crates/types/src/string.rs

```diff
@@ -251,7 +251,7 @@ impl StringBuilder {
             // terminated by a null byte.
             if s.data.is_null() {
                 unsafe {
-                    let ptr = libc::malloc(1) as *mut i8;
+                    let ptr = libc::malloc(1) as *mut std::ffi::c_char;
```

Comment:
Nit: `ffi` is already in scope I believe.

```suggestion
                    let ptr = libc::malloc(1) as *mut ffi::c_char;
```